website/blog/2025-01-15-version-0.77.md

```diff
@@ -0,0 +1,375 @@
+
+If your app falls in this category, please skip the migration of the AppDelegate to Swift, and keep using Objective-C++ for your app.
+
+React Native core is mostly developed using C++ to encourage code sharing between iOS and Android and other platforms. The interoperability between Swift and C++ is not mature nor stable, yet. We are looking into ways to fill this gap and let you migrate to Swift too.
+
+##### RCTAppDependencyProvider
+
+React Native 0.77 slightly changes how the app loads third party dependencies. This is a new line in the community template that, if missed, can cause some runtime issues. Make sure to add it to your app.
+
+The equivalent Objective-C lines are the following:
+
+```objc title=”AppDelegate.mm”
+#import "AppDelegate.h"
+
+#import <React/RCTBundleURLProvider.h>
+// highlight-next-line
+#import <ReactAppDependencyProvider/RCTAppDependencyProvider.h>
+
+
+@implementation AppDelegate
+
+- (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions
+{
+  self.moduleName = @"<Your app Name>";
+  // highlight-next-line
+  self.dependencyProvider = [RCTAppDependencyProvider new];
+  // You can add your custom initial props in the dictionary below.
+  // They will be passed down to the ViewController used by React Native.
+  self.initialProps = @{};
+
+  return [super application:application didFinishLaunchingWithOptions:launchOptions];
+}
+
+// remaining of the AppDelegate
+
+```
+
+## Breaking Changes
+
+### Removal of `console.log()` streaming in Metro
+
+We want every aspect of React Native debugging to behave reliably and to match the functionality of modern browser tooling. To meet this quality bar, log forwarding via Metro, originally deprecated in 0.76, is removed in 0.77.
+
+This integration relied on a custom approach to communicate with the debugging target on device. With this change, we are moving exclusively to the Chrome DevTools Protocol (CDP).
+
+- To view JS logs, please use [React Native DevTools](https://reactnative.dev/docs/react-native-devtools) and its fully featured Console panel — supporting log filtering, rich object inspection, Live Expressions, and more.
+- You can also connect VS Code as a CDP debugger, via third party extensions such as [Expo Tools](https://github.com/expo/vscode-expo) and [Radon IDE](https://ide.swmansion.com/).
+  - Please note that these integrations are not directly supported by the React team. However, we are working on first party VS Code support in 2025.
+- Expo continues to offer log streaming in Expo CLI.
+
+For more info, see [_Why are JavaScript logs leaving Metro?_](https://github.com/react-native-community/discussions-and-proposals/discussions/819#:~:text=Why%20are%20JavaScript%20logs%20leaving%20Metro%3F)
+
+### Other Breaking Changes
```

Comment:
What is the criteria for what appears in this section vs what we add to [the changelog](https://github.com/facebook/react-native/blob/main/CHANGELOG.md)?